libavformat/mxfdec.c

```diff
@@ -2462,70 +2467,72 @@ static MXFTrack* mxf_get_dovi_metadata_track(MXFContext* mxf)
     }
 
     if (-1 == track_id) {
-        // No interleaved Dolby Vision metadata exists
+        // No interleaved PHDR metadata exists
         return NULL;
     }
 
     // Obtain the actual MXF track containing presumed Dolby Vision metadata, based on mapped track ID
     for (size_t k = 0; k < mxf->metadata_sets_count; k++) {
         MXFMetadataSet *metadata = mxf->metadata_sets[k];
         if (metadata->type == Track && (track_id == ((MXFTrack*)(metadata))->track_id)) {
-            dovi_metadata_track = (MXFTrack*)metadata;
+            phdr_metadata_track = (MXFTrack*)metadata;
             break;
         }
     }
 
-    av_log(mxf->fc, AV_LOG_TRACE, "found in use Dolby Vision metadata track id %" PRIu64 "\n", track_id);
-    return dovi_metadata_track;
-}
-
-static int mxf_init_dovi_metadata_stream(MXFContext* mxf, AVStream* st)
-{
-    int ret = 0;
-
-    if (!mxf->dovi_global_metadata) {
-        av_log(mxf->fc, AV_LOG_TRACE, "no Dolby Vision global metadata found in metadata sets\n");
-        return AVERROR_INVALIDDATA;
-    }
-
-    ret = av_dict_set(&st->metadata, "dovi_global_metadata", mxf->dovi_global_metadata->data, 0 /* flags */);
-    return ret;
+    av_log(mxf->fc, AV_LOG_TRACE, "found in use PHDR metadata source track id %" PRIu64 "\n", track_id);
+    return phdr_metadata_track;
 }
 
-static int mxf_add_dovi_metadata_stream(MXFContext* mxf)
+static int mxf_add_phdr_metadata_stream(MXFContext* mxf)
 {
     MXFTrack* track = NULL;
     MXFStructuralComponent *component = NULL;
     const MXFCodecUL *container_ul = NULL;
     AVStream *st = NULL;
 
-    if (!(track = mxf_get_dovi_metadata_track(mxf))) {
+    if (!(track = mxf_get_phdr_metadata_track(mxf))) {
         return 0;
     }
 
+    // Validate that PHDR metadata track refers to the correct components
     if (!(track->sequence = mxf_resolve_strong_ref(mxf, &track->sequence_ref, Sequence))) {
         av_log(mxf->fc, AV_LOG_ERROR, "could not resolve track sequence strong ref\n");
         return AVERROR_INVALIDDATA;
     }
+    if (!mxf_match_uid(track->sequence->data_definition_ul, mxf_phdr_image_metadata_item,
```

Comment:
Added additional validation here to ensure all components necessary for PHDR to exist within a track are met.